### PR TITLE
Add metric query endpoint

### PR DESCRIPTION
**Pull request by kamikaz1k**
## Work items
- [x] `offset` pagination (W E B S C A L E)
- [x] datetime range query (`start_datetime` - `end_datetime`)
- [x] json-path selection of arbitrary data is _NOT_ part of this PR

Query format:
`/metrics?offset=4&start_datetime=2019-11-11T01:10:00&end_datetime=2019-11-11T01:18:00`

It works with any combination of the 3 parameters.

## Dev Notes (historical):
Originally tried to do dynamic query segments using the ORM using `Box` and `BoxableExpression` types (see [SO answer here](https://stackoverflow.com/questions/48696290/creating-diesel-rs-queries-with-a-dynamic-number-of-ands)).

It was very complicated, but moreover we found out that json operators haven't yet been implemented for `diesel` yet. See https://github.com/diesel-rs/diesel/issues/44.

So we instead are going with `sql_query(...)` to generate a raw SQL query. Luckily diesel provides a `QueryableByName` to still figure out the queried result can be directly instantiated into diesel Model types. 👌

### CODE REVIEW

**Review by woodgern — COMMENTED** [review]
Word, I'm currently looking at parsing the QL and turning it into a diesel based query

File: metrix/src/main.rs [hunk 1]
```diff
@@ -41,6 +44,73 @@ fn create_metric_route(metric_body: Json<NewMetric>) -> Json<Metric> {
...
+                    filter_clause.len(),
+                    &format!(" AND created_at < '{}'", created_at_end).to_string()
+                );
+            }
+        }
+    }
+
+    println!("### QUERY: SELECT * FROM metrics {};", filter_clause);
+
+    let query_string = format!("SELECT * FROM metrics {} ORDER BY id LIMIT 10", filter_clause);
```
Comment (woodgern) [inline metrix/src/main.rs:106]: Any reason for limit 10?

Comment (kamikaz1k) [reply metrix/src/main.rs:106]: to test pagination, we can change it, nbd.

File: metrix/src/main.rs [hunk 2]
```diff
@@ -41,6 +44,73 @@ fn create_metric_route(metric_body: Json<NewMetric>) -> Json<Metric> {
...
+            let created_at_start = result.ok().unwrap();
+            let created_at_start_parsed = NaiveDateTime::parse_from_str(
+                &created_at_start,
+                &"%Y-%m-%dT%H:%M:%S".to_string() // "2014-5-17T12:34:56"
+            );
+
+            if created_at_start_parsed.is_ok() {
+                filter_clause.insert_str(
+                    filter_clause.len(),
+                    &format!(" AND created_at > '{}'", created_at_start).to_string()
```
Comment (woodgern) [inline metrix/src/main.rs]: Might want `>=` here? Totally a 'product' thing, but I feel like 'start_date' is inclusive

File: metrix/src/main.rs [hunk 3]
```diff
@@ -41,6 +44,73 @@ fn create_metric_route(metric_body: Json<NewMetric>) -> Json<Metric> {
...
+            let created_at_end = result.ok().unwrap();
+            let created_at_end_parsed = NaiveDateTime::parse_from_str(
+                &created_at_end,
+                &"%Y-%m-%dT%H:%M:%S".to_string() // "2014-5-17T12:34:56"
+            );
+
+            if created_at_end_parsed.is_ok() {
+                filter_clause.insert_str(
+                    filter_clause.len(),
+                    &format!(" AND created_at < '{}'", created_at_end).to_string()
```
Comment (woodgern) [inline metrix/src/main.rs]: Same here (not  at all critical)

File: metrix/src/main.rs (hunk 2 above)
Comment (woodgern) [reply metrix/src/main.rs]: Not critical btw

**Review by woodgern — APPROVED** [review]
LGTM, Let's Get This Merged

**Pull request merged**